### PR TITLE
🏃 hack/setup-envtest.sh: follow-up from #1092

### PR DESCRIPTION
**Pull request by joelanford**
This follows up #1092 to address some of the comments and suggestions.

### CODE REVIEW

**Comment by joelanford** [comment]
/cc @camilamacedo86 @vincepri @alvaroaleman @DirectXMan12 

@vincepri Just saw your follow-up about copying the header text function. I'll update this PR now.

**Comment by vincepri** [comment]
Thanks for following up! The changes look great, will tag it once the header text is in

**Comment by joelanford** [comment]
@vincepri WDYT about removing `set -o nounset` from `setup-envtest.sh`?

I was testing it out by running `source ./hack/setup-envtest.sh; echo $KUBEBUILDER_ASSETS` and expecting to see it return an empty string before calling `setup_envtest_env`.

Instead, my terminal exited because `$KUBEBUILDER_ASSETS` was not set yet.

We could add this to `hack/check-everything.sh` since we don't expect people to be `source`-ing that script.

**Comment by vincepri** [comment]
Got it, I don't have time to try today, but I didn't get the same error in a similar cluster api script

File: hack/setup-envtest.sh [hunk 1]
```diff
@@ -14,32 +14,60 @@
...
+
+function header_text {
+  echo "$header$*$reset"
+}
 
 function setup_envtest_env {
+  header_text "setting up env vars"
+
   # Setup env vars
+  KUBEBUILDER_ASSETS=${KUBEBUILDER_ASSETS:-""}
```
Comment (vincepri) [inline hack/setup-envtest.sh:41]: Can probably remove this?

File: hack/setup-envtest.sh [hunk 2]
```diff
@@ -14,32 +14,60 @@
...
+# Skip fetching and untaring the tools by setting the SKIP_FETCH_TOOLS variable
+# in your environment to any value:
+#
+# $ SKIP_FETCH_TOOLS=1 ./check-everything.sh
+#
+# If you skip fetching tools, this script will use the tools already on your
+# machine.
 function fetch_envtest_tools {
+  SKIP_FETCH_TOOLS=${SKIP_FETCH_TOOLS:-""}
+  if [ -n "$SKIP_FETCH_TOOLS" ]; then
```
Comment (vincepri) [inline hack/setup-envtest.sh:58]: Can probably remove this?

File: hack/setup-envtest.sh (hunk 1 above)
Comment (joelanford) [reply hack/setup-envtest.sh:41]: Actually had to keep these if we want to use `set -o nounset` in `./hack/check-everything.sh`

**Pull request merged**